Library/ViewModels/CommentsViewModel.swift

```diff
@@ -216,13 +216,30 @@ public final class CommentsViewModel: CommentsViewModelType,
 
     let commentComposerDidSubmitText = self.commentComposerDidSubmitTextProperty.signal.skipNil()
 
-    self.failableOrComment <~ Signal.combineLatest(
-      initialProject,
-      currentUser.skipNil()
-    )
-    .takePairWhen(commentComposerDidSubmitText)
-    .map(unpack)
-    .flatMap(.concurrent(limit: concurrentCommentLimit), postCommentProducer)
+    let commentableId = projectOrUpdate
+      .flatMap { projectOrUpdate in
+        projectOrUpdate.ifLeft { project in
+          SignalProducer.init(value: project.graphID)
+        } ifRight: { update in
+          SignalProducer.init(value: encodeToBase64("FreeformPost-\(update.id)"))
```

Comment:
What is the relationship between `Update.swift` and `FreeformPost`. Can we conform to GraphIDBridging like we do in the other graph schema classes so we get the `.graphID` property for use here?

Reply:
So we can use `GraphIDBridging` however the modelName would be `FreeformPost` , the only relationship between `Update` and `FreeformPost` is that the response to fetch comments for Update is wrapped inside `FreeformPost`.